source/common/request_info/dynamic_metadata_impl.cc

```diff
@@ -0,0 +1,30 @@
+#include "common/request_info/dynamic_metadata_impl.h"
+
+#include "envoy/common/exception.h"
+
+namespace Envoy {
+namespace RequestInfo {
+
+void DynamicMetadataImpl::setData(absl::string_view data_name, std::unique_ptr<Object>&& data) {
+  if (data_storage_.find(data_name) != data_storage_.end()) {
+    throw EnvoyException("DynamicMetadata::setData<T> called twice with same name.");
+  }
+  data_storage_[static_cast<std::string>(data_name)] = std::move(data);
```

Comment:
nit: using a cast here is a bit strange to me. Is this just to force type coercion when it doesn't happen otherwise?

Reply:
Yes; absl::string_view doesn't convert to std::string unless you ask it nicely :-}.  Added a comment.